src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,159 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * Encapsulates the loading of tasks from a text file and storing of tasks in that text file.
+ * The text file is located in the data directory of the root folder.
+ */
+public class Storage {
+
+    /** Directory file. */
+    private final File directory;
+
+    /** Text file. */
+    private final File textFile;
+
+    /** UI for storage to display messages. */
+    private final Ui ui;
+
+    /** Task list for storage to access. */
+    private final TaskList taskList;
+
+    /**
+     * Creates a storage for Duke's tasks data.
+     *
+     * @param directoryPath Path of the directory.
+     * @param textFilePath Path of the text file.
+     * @param ui UI.
+     * @param taskList List of tasks.
+     */
+    public Storage(String directoryPath, String textFilePath, Ui ui, TaskList taskList) {
+        this.directory = new File(directoryPath);
+        this.textFile = new File(textFilePath);
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Gets the UI.
+     *
+     * @return UI.
+     */
+    public Ui getUi() {
+        return ui;
+    }
+
+    /**
+     * Gets the list of tasks.
+     *
+     * @return List of tasks.
+     */
+    public TaskList getTaskList() {
+        return taskList;
+    }
+
+    /**
+     * Checks if duke.txt file exists in the data directory of the root folder.
+     * If it exists, load the data into Duke.
+     * If it does not exist, create one.
+     */
+    public void checkSaveFile() {
```

Comment:
Perhaps can consider changing the method name into checkSavedFile().

Reply:
Done. Thanks!